Packs/GitGuardian/Integrations/GitGuardianEventCollector/GitGuardianEventCollector.py

```diff
@@ -0,0 +1,383 @@
+            List: A list of events that were fetched
+            str: The time to start the next incident fetch.
+        """
+        from_fetch_time = last_run.get("from_fetch_time", "")
+        to_fetch_time = last_run.get("to_fetch_time", "")
+        last_fetched_event_ids = last_run.get("last_fetched_ids", [])
+        next_url_link = last_run.get("next_url_link", "")
+
+        events, last_fetched_event_ids, next_fetch_url, is_nextTrigger = self.retrieve_events(
+            from_fetch_time, to_fetch_time, max_events_per_fetch, last_fetched_event_ids, event_type, next_url_link
+        )
+
+        if is_nextTrigger:
+            # handle the case where we do not need to update the time window for the next fetch, as we need to
+            # continue fetching more pages
+            next_run_from_fetch_time = last_run.get("from_fetch_time", "")
+        else:
+            # there are no more events to fetch in the current time window
+            next_run_from_fetch_time = last_run.get("to_fetch_time", "")
+
+        return events, last_fetched_event_ids, next_fetch_url, is_nextTrigger, next_run_from_fetch_time
+
+    def retrieve_events(
+            self, from_fetch_time: str, to_fetch_time: str, max_events_per_fetch: int, prev_run_fetched_event_ids: List[int],
+            event_type: str, next_url: str) -> tuple[List[Dict], List[int], str, bool]:
+        """retrieve events from the API
+
+        Args:
+            from_fetch_time (str): the time to start the fetch from
+            to_fetch_time (str): the time to end the fetch
+            max_events_per_fetch (int): maximum number of events to fetch in each fetch
+            prev_run_fetched_event_ids (List[int]): the ids of the events that were fetch in the last fetch that
+                                                    could be duplicated (the same time as to_fetch_time)
+            event_type (str): the type of the event.
+            get_events (bool, optional): running the function through the get-events command. Defaults to False.
+
+        """
+        events: List[dict] = []
+        params = {"from": from_fetch_time, "per_page": DEFAULT_PAGE_SIZE, "to": to_fetch_time}
+
+        while len(events) < max_events_per_fetch:
+            if next_url:
+                demisto.debug(f"GG: Fetching events using the next_url: {next_url}")
+                response = self._http_request(
+                    method="GET",
+                    full_url=next_url,
+                    retries=3,
+                )
+            else:
+                demisto.debug(f"GG: Fetching events using the params: {params}")
+                response = self._http_request(
+                    method="GET",
+                    url_suffix=EVENT_TYPE_TO_ENDPOINT.get(event_type, ''),
+                    params=params,
+                    retries=3,
+                )
+            new_events = self.remove_duplicated_events(response.get("results"), prev_run_fetched_event_ids)
```

Comment:
why would we run this on the same `prev_run_fetched_event_ids` over multiple pages?

Reply:
The API returns results consequently in the incident ID, and not in the last_occurence_date field, which means that if event 1 happened after event number 2, they will still get returned as [event 1, event 2]